hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -1547,3 +1547,53 @@ def hpat_pandas_series_nunique_impl(self, dropna=True):
             return len(data_set) + 1
 
     return hpat_pandas_series_nunique_impl
+
+
+@overload_method(SeriesType, 'median')
+def hpat_pandas_series_median(self, axis=None, skipna=True, level=None, numeric_only=None):
+    """
+    Pandas Series method :meth:`pandas.Series.median` implementation.
+
+    .. only:: developer
+
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_median1
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_median_skipna_default1
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_median_skipna_false1
+
+    Parameters
+    -----------
+    self: :obj:`pandas.Series`
+          input series
+    axis: axis for the function to be applied on, default None
+         *unsupported*
+    skipna: :obj:`bool`, default True
+        exclude NA/null values when computing the result
+    level: :obj:`int` or name
+         *unsupported*
+    numeric_only: :obj:`bool` or None, default None
+         *unsupported*
+
+    Returns
+    -------
+    :obj:`float` or :obj:`pandas.Series` (if level is specified)
+         median of values in the series
+
+    """
+
+    _func_name = 'Method median().'
+
+    if not isinstance(self, SeriesType):
+        raise TypingError(
+            '{} The object must be a pandas.series. Given self: {}'.format(_func_name, self))
+
+    if not isinstance(self.dtype, (types.Integer, types.Float)):
+        raise TypingError(
+            '{} The function only applies to elements that are all numeric. Given data type: {}'.format(_func_name, self.dtype))
+
+    def hpat_pandas_series_median_impl(self, axis=None, skipna=True, level=None, numeric_only=None):
+        if skipna:
```

Comment:
I see no checks for `skipna` type. What happened if it is `None`?

Reply:
Corrected by adding checks.